File: MinimumMovestoEqualArrayElementsII3.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    int minMoves2(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        
        int result = 0;
        int i = 0, j = nums.size() - 1;
        while(i < j) {
        	result += nums[j--] - nums[i++];
		}
		
		return result;
    }
};

int main() {
	
	int n;
	cin>>n;
	
	vector<int> nums;
	for(int i = 0; i < n; i++) {
		int num;
		cin>>num;
		nums.push_back(num);
	}
	
	Solution *solution = new Solution();
	cout<<solution->minMoves2(nums);
	
	return 0;
	
}

File: MinimumMovestoEqualArrayElementsII3.py
class Solution:
    def minMoves2(self, nums) -> int:
        nums.sort()
        n = len(nums)
        s = sum(nums)
        curSum = 0
        result = float('inf')
        for i in range(n):
            k = nums[i]
            curSum += k
            result = min(result, 2 * (i + 1) * k - n * k + s - 2 * curSum)
        return result


if __name__ == '__main__':
    nums = [int(num) for num in input().split()]
    solution = Solution()
    print(solution.minMoves2(nums))
